app/controllers/concerns/my_facilities_filtering.rb

```diff
@@ -5,38 +5,49 @@ module MyFacilitiesFiltering
   NEW_FACILITY_THRESHOLD = 3.months.ago
 
   included do
-    before_action :populate_facility_sizes, :populate_zones,
-      :set_selected_sizes, :set_selected_zones, :set_only_new_facilities
-
-    def filter_facilities(scope_namespace = [])
-      facilities = current_admin.accessible_facilities(:view_reports)
-
-      filtered_facilities = facilities_by_size(facilities)
+    before_action :populate_facility_groups
+    before_action :set_selected_facility_group
+    before_action :populate_facilities
+    before_action :populate_facility_sizes
+    before_action :populate_zones
+    before_action :set_selected_facility_sizes
+    before_action :set_selected_zones
+
+    def filter_facilities
+      filtered_facilities = facilities_by_size(@facilities)
       facilities_by_zone(filtered_facilities)
     end
 
     private
 
+    def populate_facility_groups
+      @facility_groups = current_admin.accessible_facility_groups(:view_reports).order(:name)
```

Comment:
`accessible_facility_groups` will give you the list of facility groups that a user has _full_ access to. In other words, the user can access _all_ facilities within that facility group.

However, the filters need to expose facility groups that the user has _any_ access to. In other words, facility groups where the user has _at least one_ facility access.

For example, if I have reports access to _one_ facility in District A and _one_ facility in District B, I expect to be able to select District A or District B from this dropdown. And in both cases, I expect to see data for only _one_ facility inside. Currently, when I tried to set up this permission for a test user, the page breaks, as `@facility_groups` becomes an empty array.

Reply:
Ah thanks for pointing this out @harimohanraj89! The example you mention is right — My Facilities should show all facilities an admin has access to and not just all facilities within districts the admin has full access to.

What do I need to change from the `populate_facility_groups` method to make sure it works as your example?